cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -283,6 +293,67 @@ public Future<Void> reconcile(KafkaStatus kafkaStatus, Clock clock)    {
                 .compose(i -> updateKafkaMetadataState(kafkaStatus));
     }
 
+    private Future<Void> updateKafkaAutoRebalanceStatus(KafkaStatus kafkaStatus) {
+        // we are going to save a consistent list of added notes (between the ones already in the status and the newly added ones)
+        // into the KafkaStatus autorebalance field to be passed to the auto-rebalancing reconciler.
+        // It's needed in case nodes are scaled up but reconciliation fails and on the next one the operator has to know where to rebalance
+
+        Set<Integer> addedBrokers = new LinkedHashSet<>(kafka.addedBrokerNodes());
+        if (kafkaAutoRebalanceStatus == null) {
+            // no already stored added nodes, adding only the new ones from current reconciliation
+            if (!addedBrokers.isEmpty()) {
+                kafkaStatus.setAutoRebalance(
+                        new KafkaAutoRebalanceStatusBuilder()
+                                .withModes(
+                                        new KafkaAutoRebalanceModeBrokersBuilder()
+                                                .withMode(KafkaRebalanceMode.ADD_BROKERS)
+                                                .withBrokers(kafka.addedBrokerNodes().stream().toList())
+                                                .build()
+                                )
+                                .build());
+            }
+            return Future.succeededFuture();
+        } else {
+            KafkaAutoRebalanceStatusBuilder builder = new KafkaAutoRebalanceStatusBuilder(kafkaAutoRebalanceStatus);
+            // no already stored added nodes, adding only the new ones from current reconciliation
```

Comment:
Should this comment be under the next if? It would seem to make more sense given the next comment is after the else?